packages/example-app-base/src/components/SideRail/SideRail.tsx

```diff
@@ -0,0 +1,186 @@
+
+    if (!jumpLinks || !jumpLinks.length) {
+      return null;
+    }
+    const links = jumpLinks.map((jumpLink: ISideRailLink) => (
+      <li
+        key={jumpLink.url}
+        className={css(classNames.link, classNames.jumpLink, jumpLink.url === activeLink && sideRailClassNames.isActive)}
+      >
+        <Link href={this._getPagePath(jumpLink.url)} data-anchor-url={jumpLink.url} onClick={this._onJumpLinkClick}>
+          {jumpLink.text}
+        </Link>
+      </li>
+    ));
+    return (
+      <div className={css(classNames.section, classNames.jumpLinkSection)}>
+        <h3 className={classNames.sectionTitle}>On this page</h3>
+        <ul>{links}</ul>
+      </div>
+    );
+  };
+
+  private _renderRelatedLinkList = (): JSX.Element | null => {
+    const { relatedLinks } = this.props;
+    const classNames = this._classNames;
+
+    let links: JSX.Element | undefined;
+    if (_isElement(relatedLinks)) {
+      links = <div className={classNames.markdownList}>{relatedLinks}</div>;
+    } else if (Array.isArray(relatedLinks) && relatedLinks.length) {
+      links = (
+        <ul>
+          {relatedLinks.map(
+            (relatedLink: ISideRailLink) =>
+              !isPageActive(relatedLink.url) && (
+                <li key={relatedLink.url} className={classNames.link}>
+                  <Link href={relatedLink.url}>{relatedLink.text}</Link>
+                </li>
+              )
+          )}
+        </ul>
+      );
+    }
+
+    if (links) {
+      return (
+        <div className={css(classNames.section)}>
+          <h3 className={classNames.sectionTitle}>Related pages</h3>
+          {links}
+        </div>
+      );
+    }
+    return null;
+  };
+
+  private _renderContactList = (): JSX.Element | null => {
+    const { contactLinks } = this.props;
+    const classNames = this._classNames;
+
+    let links: JSX.Element | undefined;
+    if (_isElement(contactLinks)) {
+      links = <div className={classNames.markdownList}>{contactLinks}</div>;
+    } else if (Array.isArray(contactLinks) && contactLinks.length) {
+      links = (
+        <ul>
+          {contactLinks.map((contactLink: ISideRailLink) => (
+            <li key={contactLink.url} className={classNames.link}>
+              <Link href={contactLink.url}>{contactLink.text}</Link>
+            </li>
+          ))}
+        </ul>
+      );
+    }
+
+    if (links) {
+      return (
+        <div className={css(classNames.section)}>
+          <h3 className={classNames.sectionTitle}>Contacts</h3>
+          {links}
+        </div>
+      );
+    }
+
+    return null;
+  };
+
+  // tslint:disable-next-line:no-any
+  private _onJumpLinkClick = (ev?: React.MouseEvent<any>): void => {
+    const target = ev && (ev.target as HTMLAnchorElement);
+    const url = target && target.dataset && target.dataset.anchorUrl;
+    if (url && ev) {
+      ev.preventDefault();
+    }
+    return jumpToAnchor(url);
+  };
+
+  private _getPagePath(url: string): string {
```

Comment:
Would be nice to have the variables renamed in this function, or at least some example inputs->outputs. There are several things that seem odd (the `path` is the bit _before_ the `hash`, putting a url after a `#` is also strange).